### ISSUE TITLE
Parity auto-update downloads broken package (possibly build for another system)

### ISSUE THREAD

**Issue by 0v1se** [paritytech/parity-ethereum#8816]
> I'm running:
>
> - **Which Parity version?**: 1.10.4
> - **Which operating system?**: Linux (CentOS Linux release 7.3.1611 (Core))
> - **How installed?**: binaries (rpm for centos)
> - **Are you fully synchronized?**: yes
> - **Which network are you connected to?**: ethereum 
> - **Did you try to restart the node?**: yes

On the last auto-update incorrect binary was downloaded. Here is the log:
```
2018-06-06 03:17:52  Already fetched binary.
2018-06-06 03:17:52  Completed upgrade to 1.10.6-stable-bc0d…2c8d
2018-06-06 03:17:52  Finishing work, please wait...
2018-06-06 03:17:52  Imported #5739162 0556…8b2e (202 txs, 7.99 Mgas, 439.41 ms, 30.69 KiB)
/home/ether/.local/share/io.parity.ethereum-updates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d: /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /home/ether/.local/share/io.parity.ethereum-upda
tes/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d)
/home/ether/.local/share/io.parity.ethereum-updates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/ether/.local/share/io.parity.ethereum-up
dates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d)
/home/ether/.local/share/io.parity.ethereum-updates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /home/ether/.local/share/io.parity.ethereum-up
dates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d)
/home/ether/.local/share/io.parity.ethereum-updates/parity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /home/ether/.local/share/io.parity.ethereum-updates/par
ity-1.10.6-bc0d134bb43c65a5b82bd18c311319c7036d2c8d)
```

Resolved this issue by deleting .local/share/io.parity.ethereum-updates folder and updating manually using rpm for centos.

**Comment by tomusdrw** [comment]
I suppose we should only enable updater on supported distros, the version string for unsupported platform is most likely incorrect.

**Comment by luarx** [comment]
I don't understand why a tagged Docker image can update the Parity version if it wants! 
**IT DOESN'T MAKE SENSE** I think...

As I understand, I can be using image Docker image v1.10.4 but internally this image is using Parity v1.11...
Yes, it is true that with "auto-update=none" parameter it can be disabled, but by default it's set to "critical". 

Finally, another thing is that if an update is critical, why is it not testing correctly!?

**Comment by tomusdrw** [comment]
@luarx Parity is upgrading to critical updates by default, as you already know you can disable that by specifiying `auto-update=none`.
The update happens only within a release track, so `stable` release will only update to another `stable` release (i.e. 1.9-stable can update to 1.10-stable), 1.11 is now in beta, so it'd only update if you were running (rather old) 1.10-beta series.

Critical updates are used sparingly, and only to prevent consensus issues that could bring the entire network down. If you don't update you will most likely not be able to use that node anyway, because of incompatibility.

Regarding this thread - afair Docker images are not built on CentOs (which is reported in this thread) and Ubuntu/Debian installations should update correctly.

That said we are working on improving auto-updater to prevent shutting down your node in case of failure, see #8178 

If you have experienced another issue with Parity and/or parity docker image, please open a separate issue.

**Comment by luarx** [comment]
@tomusdrw as I understand auto updating system by default (critical updates only) works like:
- `Non stable release` --update to-->` Last non stable release`. Example: v1.10.2-beta ---> v1.11.3-beta
- `Stable release` --update to--> `Last stable release`. Example: v1.10.3-stable ---> v1.10.6-stable. Because v1.10.6 is the last stable version and, also, v1.10.4-stable won't be used automatically because it is not critical update.

Is it correct?

Yes, I understand that critical updates should be done if they could bring the entire network down...but in my opinion, Docker images are like box which has to be the same wherever and whenever you execute it unless you use a non versioned image tag as :latest. So as I was using a versioned docker image I was confident that it will work forever.

Finally, would it be a good feature to tag docker stable images like "v1.10.6-stable"? So that it will be easy for everybody to know what images are stable without going to github releases page.

**Comment by 5chdn** [comment]
We need to give this some thoughts as we are reducing the number of supported distros significantly soon.

**Comment by holiman** [comment]
This happens for me also on docker, using `parity/parity:latest`
```
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:23 UTC Update for binary 0xd095…d9fb triggered 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:23 UTC Attempting to get parity binary 0xd095…d9fb 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:23 UTC Imported #5746840 0xd0a6…d1be (160 txs, 8.00 Mgas, 356.31 ms, 22.72 KiB) 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:34 UTC Fetched latest version (1.11.3-beta-0xa66e…77f3) OK to /tmp/ZcVO2RZI1rrK 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:35 UTC Copied updated binary to /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:35 UTC Completed upgrade to 1.11.3-beta-0xa66e…77f3 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:35 UTC Finishing work, please wait... 
Jun 07 11:31:04 mon03.ethdevops.io parity: 2018-06-07 08:55:42 UTC Imported #5746841 0x3dbd…37f6 (205 txs, 7.99 Mgas, 373.12 ms, 30.49 KiB) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
Jun 07 11:31:04 mon03.ethdevops.io parity: /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /root/.local/share/io.parity.ethereum-updates/parity-1.11.3-a66e36bf41f9b251cb9e7c70b8fb46754beb77f3) 
```

**Comment by luarx** [comment]
@jasonOramo @holiman Have you tried to execute Parity with the parameter `--auto-update=none`?

**Comment by holiman** [comment]
@luarx good idea, will try it

**Comment by Tbaut** [comment]
@jasonOramo what do you mean "it doesn't work"  please be more precise for us to be able to help.

**Comment by holiman** [comment]
`--auto-update=none` works for me. Though, a bit surprising that the docker image does not seem to be updated automatically to the most recent version. I can recommend using docker hub's integrated builds, that's what go-ethereum uses.

**Comment by jasonOramo** [comment]
@luarx I install the version 1.10.6 manually and it works this morning. But now it hangs and I kill it. 
After I start parity again, I get this issue https://github.com/paritytech/parity/issues/8833. And all the things stuck here. No ideas about what I can do.

**Comment by joshua-mir** [comment]
closing as stale, likely resolved by #8787 and overall changes in how we distribute parity
